thoth/package_extract/image.py

```diff
@@ -546,6 +546,50 @@ def get_image_size(path: str) -> int:
     return total_size
 
 
+def _get_cuda_version(path: str) -> dict:
+    """Get the cuda version."""
+    res = dict()
+    # gathering version from version.txt file
+    version_path = path + "/usr/local/cuda/version.txt"
+    if os.path.isfile(version_path):
+        with open(version_path, "r") as f:
+            for line in f.readlines():
+                if line.startswith("CUDA Version"):
+                    res["/usr/local/cuda/version.txt"] = line[len("CUDA Version"):].strip()
+                    _LOGGER.info("CUDA version identifier was found in file %s", res["/usr/local/cuda/version.txt"])
+                    break
+                else:
+                    _LOGGER.debug("No CUDA version identifier was found in file %r", version_path[len(path):])
+    else:
+        _LOGGER.warning("No %r file was found", version_path[len(path):])
+
+    # gathering version from nvcc command
+    nvcc_path = glob.glob(path + '/usr/local/cuda/bin/nvcc')
```

Comment:
> https://github.com/thoth-station/package-extract/pull/307#discussion_r356039865
So, I investigated the image and found out that there is going to be `cuda` symlink which will link to default `cuda folder with specific version`, like in this case `cuda` refers to `cuda-10.1`.
```
root@c3624a58d7d4:/usr/local# pwd
/usr/local
root@c3624a58d7d4:/usr/local# ls -l
total 36
drwxr-xr-x. 2 root root 4096 Aug  7 13:02 bin
lrwxrwxrwx. 1 root root    9 Aug 26 21:23 cuda -> cuda-10.1
drwxr-xr-x. 1 root root 4096 Aug 26 21:26 cuda-10.1
drwxr-xr-x. 2 root root 4096 Aug  7 13:02 etc
drwxr-xr-x. 2 root root 4096 Aug  7 13:02 games
drwxr-xr-x. 2 root root 4096 Aug  7 13:02 include
drwxr-xr-x. 2 root root 4096 Aug  7 13:02 lib
lrwxrwxrwx. 1 root root    9 Aug  7 13:02 man -> share/man
drwxr-xr-x. 2 root root 4096 Aug  7 13:03 sbin
drwxr-xr-x. 1 root root 4096 Aug 26 21:22 share
drwxr-xr-x. 2 root root 4096 Aug  7 13:02 src
root@c3624a58d7d4:/usr/local# 
```
So, I think maybe it is better to check this path `/usr/local/cuda/bin/nvcc` instead of `/usr/local/cuda-*/bin/nvcc` how I did previously. WDYT?
And, I think we can just get the default cuda version and make it simple. I am pretty sure that most of the images will come with one cuda version. Yes, multiple cuda versions could be installed on real machines, but when it comes to images, it is so rare or even not a case. WDYT?

Reply:
Nice! Definitely makes sense! :+1:

Reply:
Definitely makes sense! Nice! :+1: 

Now you don't need to glob, having just `os.path.join` will work fine

Reply:
Thanks, done!